client/src/routes/water-meter/WaterMeterDataSet.js

```diff
@@ -0,0 +1,88 @@
+import moment from 'moment';
+import _ from 'lodash';
```

Comment:
Sure, I'm going to ask you how to do it correctly at the lesson

Reply:
Imports:
```import fpFilter from 'lodash/fp/filter';
import flow from 'lodash/flow';
import fpGroupBy from 'lodash/fp/groupBy';
import fpMap from 'lodash/fp/map';```
Actual code:
```filteredEvents = flow(
    fpFilter(({ Created }) =>
      moment(Created).isAfter(moment().startOf(period)) &&
      moment(Created).isBefore(moment().endOf(period)),
    fpMap(item => {
      return {
        Consumption: parseInt(item.Consumption),
        Created: moment(item.Created)
          .startOf(sortPeriod)
          .toISOString(),
      };
    }),
    fpGroupBy('Created'),
    fpMap((item, date) => ({
      Consumption: _.sumBy(item, 'Consumption'),
      Created: date,
    })),
)(filteredEvents);```

This may be too complicated for you and can be incorrect in some imports, but this is the way I use lodash nowadays